pkg/ccl/backupccl/restore_job.go

```diff
@@ -289,53 +280,58 @@ func restore(
 	if err != nil {
 		return emptyRowCount, err
 	}
-	importSpans, err := makeSimpleImportSpans(
-		dataToRestore.getSpans(),
-		backupManifests,
-		layerToBackupManifestFileIterFactory,
-		backupLocalityMap,
-		introducedSpanFrontier,
-		highWaterMark,
-		targetRestoreSpanSize.Get(execCtx.ExecCfg().SV()))
-	if err != nil {
-		return emptyRowCount, err
-	}
 
-	if len(importSpans) == 0 {
-		// There are no files to restore.
-		return emptyRowCount, nil
-	}
-
-	for i := range importSpans {
-		importSpans[i].ProgressIdx = int64(i)
+	mu := struct {
+		syncutil.Mutex
+		highWaterMark     int64
+		ceiling           int64
+		res               roachpb.RowCount
+		importSpans       map[int64]roachpb.Span
```

Comment:
at first I got a little confused thinking this was all the import spans, and thus would oom, and seeing progress reports have their key compared to its content made me think "oh yeah, it must be holding all the import spans, that is scary."

but then I think I realized on closer reading that it is only "in-flight" import spans: we generate some by pulling from the channel until our ceiling is high enough to match any progress we've heard, then we compare that our ith span's key matches the reported complete i'th span's reported key, and then we empty out that map entry, so the map should only have entires for unreported spans below the max-reported span? is that right?

if so a comment, and maybe naming this something like inflightImportSpans or something, might make this more obvious?

Reply:
Yeah that's right, once all spans up to index `N` are completed, then the highWaterMark will be moved up to `N` and everything below `N` will be removed from both maps.

I've done the rename and added a comment describing this behavior.

Reply:
Thanks!